### PR TITLE
Add the ability to use a different Jira project depending on Incident…

### PR DESCRIPTION
**Pull request by willbengtson**
Adds the ability to use a different Jira depending on the incident type. Adds a JSON column to the `incident_type` DB table which allows you to place JSON data for the given plugin. The JSON first level keys are plugin slug.

Example data:

```
{
  "dispatch-jira-plugin": {
    "project_key": "TEST123",
    "issue_type_id": 10244
  }
}
```

Also add a name to the unique constraint from the previous db migration.

### CODE REVIEW

File: src/dispatch/incident_type/models.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class IncidentType(Base):
     slug = Column(String)
     description = Column(String)
     visibility = Column(String, default=Visibility.open)
+    plugin_metadata = Column(JSON)
```
Comment (kevgliss) [inline src/dispatch/incident_type/models.py:18]: If we set the default to be an empty dict via `plugin_metadata = db.Column(JSON, nullable=False, default=dict, server_default='{}')` I think we can remove a lot of checks for `None` and returning empty dicts.

File: src/dispatch/plugins/dispatch_jira/plugin.py [hunk 2]
```diff
@@ -222,6 +223,7 @@ def update(
         storage_weblink: str = None,
         labels: List[str] = None,
         cost: int = None,
+        incident_type_plugin_metadata: dict = {},
```
Comment (kevgliss) [inline src/dispatch/plugins/dispatch_jira/plugin.py:226]: Is this still a WIP? Looks like we haven't let Jira read the incident metadata.

**Comment by willbengtson** [comment]
Updated to default dict `{}` for `plugin_metadata`.  Will PR generic Jira plugin in another PR, does that work?

File: src/dispatch/incident_type/models.py (hunk 1 above)
Comment (willbengtson) [reply src/dispatch/incident_type/models.py]: updated

File: src/dispatch/plugins/dispatch_jira/plugin.py (hunk 2 above)
Comment (willbengtson) [reply src/dispatch/plugins/dispatch_jira/plugin.py:226]: Can port generic Jira plugin in another PR?

File: src/dispatch/incident/flows.py [hunk 3]
```diff
@@ -650,6 +659,10 @@ def incident_create_flow(*, incident_id: int, checkpoint: str = None, db_session
     set_conversation_topic(incident)
 
     # we update the incident ticket
+    incident_type_data = incident_type_service.get_by_name(
+        db_session=db_session, name=incident.incident_type.name
+    )
+
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py:665]: Can we move it inside `update_incident_ticket`, so we don't have to add this code to all the places where we call `update_incident_ticket`? Same for when we call `create_incident_ticket`.

File: src/dispatch/alembic/versions/9b4a5ff08170_.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+"""empty message
```
[the remark this answers is not in the document]
Comment (willbengtson) [reply src/dispatch/alembic/versions/9b4a5ff08170_.py:1]: Dang I always forget this :(

[the remark this answers is not in the document]
Comment (mvilanova) [reply src/dispatch/alembic/versions/9b4a5ff08170_.py:1]: No worries :)

File: src/dispatch/incident/flows.py (hunk 3 above)
Comment (willbengtson) [reply src/dispatch/incident/flows.py]: updated. Had to add passing in the incident type name in a few places.

File: src/dispatch/alembic/versions/9b4a5ff08170_.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (willbengtson) [reply src/dispatch/alembic/versions/9b4a5ff08170_.py]: updated

File: src/dispatch/incident/flows.py [hunk 5]
```diff
@@ -499,6 +510,10 @@ def incident_create_flow(*, incident_id: int, checkpoint: str = None, db_session
     )
 
     # create the incident ticket
+    incident_type_data = incident_type_service.get_by_name(
+        db_session=db_session, name=incident.incident_type.name
+    )
+
```
Comment (mvilanova) [inline src/dispatch/incident/flows.py]: Forgot to remove this one?

Comment (willbengtson) [reply src/dispatch/incident/flows.py]: yeah I updated it

Comment (mvilanova) [reply src/dispatch/incident/flows.py]: All good. Looks like I wasn't looking at the latest version. Thanks

File: src/dispatch/plugins/dispatch_jira/plugin.py (hunk 2 above)
Comment (kevgliss) [reply src/dispatch/plugins/dispatch_jira/plugin.py:226]: Separate PR is fine.

**Pull request merged**